mteb/abstasks/mteb_result.py

```diff
@@ -0,0 +1,183 @@
+from __future__ import annotations
+
+import json
+from functools import lru_cache
+from pathlib import Path
+from typing import Any, Callable, Dict, Type
+
+import numpy as np
+from pydantic import BaseModel, field_validator
+from TaskMetadata import ISO_LANGUAGE, ISO_LANGUAGE_SCRIPT, LANGUAGES, HFSubset
+
+from mteb.abstasks import AbsTask
+from mteb.abstasks.languages import LangScriptFilter
+from mteb.overview import create_task_list
+
+Split = str
+Scores = Dict[str, Any]
+
+
+@lru_cache(maxsize=None)
+def create_name_to_task_mapping() -> dict[str, Type[AbsTask]]:
+    tasks = create_task_list()
+    return {cls.metadata.name: cls for cls in tasks}
+
+
+def eval_langs_as_dict(
+    eval_langs: LANGUAGES,
+) -> dict[HFSubset, list[ISO_LANGUAGE_SCRIPT]]:
+    if isinstance(eval_langs, list):
+        return {"": eval_langs}
+    return eval_langs  # type: ignore
+
+
+class MTEBResults(BaseModel):
+    """A class to represent the MTEB result.
+
+    Attributes:
+        task_name: The name of the MTEB task.
+        dataset_revision: The revision dataset for the task on HuggingFace dataset hub.
+        mteb_version: The version of the MTEB used to evaluate the model.
+        scores: The scores of the model on the dataset. The scores is a dictionary with the following structure; dict[Split, dict[HFSubset, Scores]].
+            Where Scores is a dictionary with the following structure; dict[str, Any]. Where the keys and values are scores. Split is the split of
+            the dataset. HFSubset is the subset name on huggingface e.g. "en-de". HFSubset is set to "" if there is no subsets on huggingface.
+
+    Example:
+        >>> scores = {
+        ...     "train": {
+        ...         "en-de": {
+        ...             "main_score": 0.5,
+        ...             "evaluation_time": 100,
+        ...         },
+        ...         "en-fr": {
+        ...             "main_score": 0.6,
+        ...             "evaluation_time": 200,
+        ...         },
+        ...     },
+        ... }
+        >>> mteb_results = MTEBResults(
+        ...     dataset_revision="1.0",
+        ...     task_name="sample_task",
+        ...     mteb_version="1.0",
+        ...     scores=scores,
+        ... )
+        >>> mteb_results.get_main_score()  # get the main score for all languages
+        0.55
+        >>> mteb_results.get_main_score(languages=["fra"])  # get the main score for French
+        0.6
+    """
+
+    dataset_revision: str
+    task_name: str
+    mteb_version: str
+    scores: dict[Split, dict[HFSubset, Scores]]
+
+    @field_validator("scores")
+    def _validate_scores(
```

Comment:
I know it might be a bit late for this, but maybe we should consider pulling in pydantic for all of this. Seems like a lot of validation code to write and maintain

Reply:
This is pydantic

Reply:
oh alright, ignore it then xd